h2/settings.py

```diff
@@ -8,58 +8,139 @@
 state of the settings and the unacknowledged future values of the settings.
 """
 import collections
+import enum
 
 from hyperframe.frame import SettingsFrame
 
 from h2.errors import ErrorCodes
 from h2.exceptions import InvalidSettingsValueError
 
 
+class SettingCodes(enum.IntEnum):
+    """
+        All known HTTP/2 setting codes.
+
+        .. versionadded:: 2.5.3
```

Comment:
This docstring can be dedented.

As to the versionadded comment, this won't be able to make it into the 2.5 release because this is a public part of the API, and so will need to go into 2.6. So the version added is 2.6.0.